💬 feat: assistant conversation starter

berry-13 opened the pull request:
## Summary



## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.
- [x] A pull request for updating the documentation has been submitted.

berry-13 committed:
feat: initial UI convoStart

berry-13 committed:
fix: ConvoStarter UI

berry-13 committed:
fix: convoStarters bug

berry-13 committed:
feat: Add input field focus on conversation starters

berry-13 committed:
style: conversation starter UI update

berry-13 committed:
feat: apply fixes for starters

berry-13 committed:
style: update conversationStarters UI and fixed typo

berry-13 committed:
general UI update

berry-13 committed:
feat: Add onClick functionality to ConvoStarter component

berry-13 committed:
fix: quick fix test

berry-13 committed:
fix(AssistantSelect): remove object check

berry-13 committed:
fix: updateAssistant `conversation_starters` var

danny-avila committed:
chore: remove starter autofocus

danny-avila committed:
fix: no empty conversation starters, always show input, use Constants value for max count

danny-avila committed:
style: Update defaultTextPropsLabel styles, for a11y placeholder

danny-avila committed:
refactor: Update ConvoStarter component styles and class names for a11y and theme

danny-avila committed:
refactor: convostarter, move plus button to within persistent element

danny-avila commented on client/src/components/Chat/Landing.tsx:
```diff
@@ -64,14 +69,14 @@ export default function Landing({ Header }: { Header?: ReactNode }) {
                   {assistantName}
                 </div>
-                <div className="text-token-text-secondary max-w-md text-center text-xl font-normal ">
+                <div className="text-token-text-primary max-w-md text-center text-sm font-normal ">
```
@berry-13 we need to remove `text-token-text-primary` for `text-text-primary`

danny-avila committed:
fix: types

danny-avila committed:
chore: Update landing page assistant description styling with theming

danny-avila committed:
chore: assistant types

danny-avila committed:
refactor: documents routes

danny-avila committed:
refactor: optimize conversation starter mutations/queries

danny-avila committed:
refactor: Update listAllAssistants return type to Promise<Array<Assistant>>

danny-avila committed:
feat: edit existing starters

danny-avila committed:
feat(convo-starters): enhance ConvoStarter component and add animations

    - Update ConvoStarter component styling for better visual appeal
    - Implement fade-in animation for smoother appearance
    - Add hover effect with background color change
    - Improve text overflow handling with line-clamp and text-balance
    - Ensure responsive design for various screen sizes

danny-avila committed:
feat(assistant): add conversation starters to assistant builder

- Add localization strings for conversation starters
- Update mobile.css with shake animation for max starters reached
- Enhance user experience with tooltips and dynamic input handling

danny-avila committed:
refactor: select specific fields for assistant documents fetch

danny-avila committed:
refactor: remove endpoint query key, fetch all assistant docs for now, add conversation_starters to v1 methods

danny-avila committed:
refactor: add document filters based on endpoint config

danny-avila committed:
fix: starters not applied during creation

danny-avila committed:
refactor: update AssistantSelect component to handle undefined lastSelectedModels
